### PR TITLE
[PoC] Case sensitive tests

### PR DESCRIPTION
**Pull request by phansys**
See https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1395#issuecomment-815827140.

### CODE REVIEW

File: tests/Functional/DatagridTest.php [hunk 1]
```diff
@@ -30,4 +30,32 @@ public function testFilter(): void
...
+        $this->client->request(Request::METHOD_GET, '/admin/tests/app/author/list');
+
+        $this->client->clickLink('Filters');
+        $this->client->clickLink('Address Street');
+
+        $this->client->submitForm('Filter', [
+            'filter[address__street][value]' => 'mancha',
+        ]);
+
+        self::assertSelectorTextNotContains('.sonata-link-identifier', 'Miguel de Cervantes');
```
Comment (phansys) [inline tests/Functional/DatagridTest.php:59]: Seems that we aren't getting case sensitive results with SQLite: https://www.sqlite.org/pragma.html#pragma_case_sensitive_like.

**Comment by phansys** [comment]
After some tests, I'm thinking if the required effort worths about the case sensitivity.
By instance, in order to have a concrete idea if the comparisons are case sensitive or not in MySQL, we should detect if the compared columns are using the collation `utf8mb4_0900_as_cs` or `utf8mb4_bin`, otherwise (with the current implementation) we have to detect if the `DoctrineExtensions\Query\Mysql\Binary` (`BINARY()`) DQL function is registered under the "binary" name, to allow the comparison to use that function.
A similar case is present here in SQLite, where the `PRAGMA case_sensitive_like` directive controls the behavior.
I haven't more evidence about other RDMSs, but I guess those examples are enough to bring a clear idea about the problem.
Could we re-evaluate this suggestion? https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/1387#issuecomment-812917465

**Comment by VincentLanglet** [comment]
Should we close this @phansys ?

**Comment by phansys** [comment]
Sure, let's close.

**Pull request closed without merging**